### ISSUE TITLE
Bun crashes while using standalone NextJs Build with middleware

### ISSUE THREAD

**Issue by SpectralAngel** [oven-sh/bun#14496]
### What version of Bun is running?

1.1.30

### What platform is your computer?

Docker image oven/bun:alpine

### What steps can reproduce the bug?

Using Next.js 14.2.15 and `next-auth@5.0.0-beta.22` bun crashes when the middleware is touched.

This was working with bun version 1.1.29

### What is the expected behavior?

Serving pages normally

### What do you see instead?
```
  ▲ Next.js 14.2.15
  - Local:        http://localhost:3000
  - Network:      http://0.0.0.0:3000
 ✓ Starting...
 ✓ Ready in 83ms
 ⨯  7 |   Read more: https://nextjs.org/docs/messages/middleware-new-signature
 8 |   `)}}class n extends Error{constructor(){super(`The request.page has been deprecated in favour of \`URLPattern\`.
 9 |   Read more: https://nextjs.org/docs/messages/middleware-request-page
10 |   `)}}class i extends Error{constructor(){super(`The request.ua has been removed in favour of \`userAgent\` function.
11 |   Read more: https://nextjs.org/docs/messages/middleware-parse-user-agent
12 |   `)}}},11658:(e,t,r)=>{"use strict";Object.defineProperty(t,"__esModule",{value:!0}),function(e,t){for(var r in t)Object.defineProperty(e,r,{enumerable:!0,get:t[r]})}(t,{ImageResponse:function(){return n.ImageResponse},NextRequest:function(){return i.NextRequest},NextResponse:function(){return o.NextResponse},URLPattern:function(){return s.URLPattern},userAgent:function(){return a.userAgent},userAgentFromString:function(){return a.userAgentFromString}});let n=r(65949),i=r(79789),o=r(53780),a=r(14091),s=r(88847)},45693:(e,t,r)=>{"use strict";Object.defineProperty(t,"__esModule",{value:!0}),Object.defineProperty(t,"NextURL",{enumerable:!0,get:function(){return u}});let n=r(96900),i=r(72084),o=r(57352),a=r(42150),s=/(?!^https?:\/\/)(127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}|\[::1\]|localhost)/;function c(e,t){return new URL(String(e).replace(s,"localhost"),t&&String(t).replace(s,"localhost"))}let l=Symbol("NextURLInternal");class u{constructor(e,t,r){let n,i;"object"==typeof t&&"pathname"in t||"strin
TypeError: The Request.signal getter can only be used on instances of Request
      at new c (/app/.next/server/chunks/780.js:12:9511)
      at i4 (/app/.next/server/chunks/780.js:379:51587)
      at t (/app/.next/server/chunks/780.js:379:57541)
      at /app/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:36962
      at /app/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:29696
      at /app/node_modules/next/dist/server/lib/trace/tracer.js:140:36
      at run (node:async_hooks:82:6)
      at wrap (/app/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:23166)
      at run (node:async_hooks:82:6)
      at wrap (/app/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:20514)
      at run (node:async_hooks:82:6)
      at /app/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:27581
      at execute (/app/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:27337)
      at /app/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:38301
      at handle (/app/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:38278)
      at /app/node_modules/next/dist/server/base-server.js:1352:60
      at doRender (/app/node_modules/next/dist/server/base-server.js:1264:35)
      at /app/node_modules/next/dist/server/base-server.js:1574:34
      at /app/node_modules/next/dist/server/base-server.js:1482:77
      at get (/app/node_modules/next/dist/server/response-cache/index.js:46:15)
      at /app/node_modules/next/dist/server/base-server.js:1482:53
```

**Comment by RiskyMH** [comment]
This next-auth keeps on giving me pain for trying to set it up, do you have a basic git repo that I can download and try to run instead (inc dockerfile)?

Also just to check, this works correctly with node right?

**Comment by SpectralAngel** [comment]
@RiskyMH yes this runs with node and with bun 1.1.29, I will upload a basic repo with the minimun required to make it crash

**Comment by RiskyMH** [comment]
Thanks for confirming! I can now reproduce this crash too. The next step is to figure out the root cause, but thank you for reporting it and letting us know. It definitely helps to have this reproducible case to work with.

**Comment by arno-fukuda** [comment]
Also ran into this today.
Same on oven/bun:canary-alpine

**Comment by thakuad** [comment]
Because of this i need to move back to npm for deployment through docker.

**Comment by RiskyMH** [comment]
@thakuad I don't see any reason why Bun can't be used to install dependencies, even if it's not the builder or runtime. 

Was the problem because Bun was being used as the package manager, or was it possibly due to Node.js not being installed, causing Bun to be used for runtime instead?

**Comment by alex-ketch** [comment]
> Was the problem because Bun was being used as the package manager, or was it possibly due to Node.js not being installed, causing Bun to be used for runtime instead?

@RiskyMH As another datapoint, I encountered the same issue while explicitly trying to use Bun as the runtime (`bun --bun run ...`). This was because I needed to use `bun:sqlite` in my project.
In the end I ended up returning to `@libsql/client` and Node.js runtime to get things working.

**Comment by dmeijboom** [comment]
Running `git-bisect` shows it's broken in https://github.com/oven-sh/bun/commit/a234e067a5dc7837602df3fb5489e826920cc65a. I'm not an expert in Zig (that's an understatement..) but I would assume it's this part of the change that caused this error:

```zig
if (args.getOwnTruthy(globalThis, "signal")) |signal_val| {
```

to

```zig
if (args.getTruthy(globalThis, "signal")) |signal_val| {
```

I would assume https://github.com/oven-sh/bun/issues/12368 is related based on the error but I'm not sure if that's the case.

**Comment by Rick-VA** [comment]
The issue appears to be resolved in 27067d2a6db7e85dfbd6abd6cbe18efc3ba80da2.

**Comment by Jarred-Sumner** [comment]
Fixed in 27067d2a6db7e85dfbd6abd6cbe18efc3ba80da2

This will be part of the Bun v1.1.35 release
